internal/boxcli/midcobra/telemetry.go

```diff
@@ -217,3 +206,32 @@ func (m *telemetryMiddleware) trackEvent(evt *event) {
 		UserId: evt.UserID,
 	})
 }
+
+func getSubcommand(c *cobra.Command, args []string) (subcmd *cobra.Command, subargs []string, err error) {
+	if c.TraverseChildren {
+		subcmd, subargs, err = c.Traverse(args)
+	} else {
+		subcmd, subargs, err = c.Find(args)
+	}
+	return subcmd, subargs, err
+}
+
+func getPackagesAndCommitHash(c *cobra.Command) ([]string, string) {
+	configFlag := c.Flag("config")
+	// for shell, run, and add command, path can be set via --config
+	// if --config is not set, default to current directory which is ""
+	// the only exception is the init command, for the path can be set with args
+	// since after running init there will be no packages set in devbox.json
```

Comment:
unless we run `devbox init` on a project that already has a devbox.json (we do this for the gotty-vm, see start-gotty.sh in axiom) and that fails....

Reply:
hmm i see.. well I didn't change anything here, just moved it (sorry!)